backend/models/user.js

```diff
@@ -452,29 +466,37 @@ schema.statics.getForgotPasswordToken = function(username, email, tokenExpiryTim
 		expiredAt: expiryAt
 	};
 
-	return this.findByUserName(username).then(user => {
+	let resetPasswordUserInfo = {};
 
-		if(!user) {
-			return Promise.reject(responseCodes.USER_EMAIL_NOT_MATCH);
-		}
+	let user;
+
+	return this.findUsernameOrEmail(userNameOrEmail).then(_user => {
+
+		user = _user;
 
-		if(user.customData.email !== email) {
-			return Promise.reject({ resCode: responseCodes.USER_EMAIL_NOT_MATCH});
+		if (!user) {
+			return Promise.reject(responseCodes.INVALID_USERNAME);
```

Comment:
I'm wondering if we want to tell the user that they've entered an invalid username (or e-mail address).

Maybe it should always show "Thank you. You will receive an e-mail...." even if the username/e-mail is invalid. This is so people can't figure out which users exist on our system by just using the forgot password form. Internally, we'll still have to figure out if the username/e-mail is valid so we don't try to generate a token for a non-existent user.

Reply:
I have removed all the error messages that appear , when entering an incorrect username or email address.
I have also changed the server side validation to only set a token if a username or email has been found.